File: uva_10066.py
num=1
while True:
    l1, l2=[], []
    N1, N2=input().split()
    N1, N2=int(N1), int(N2)
    if not N1 or not N2:
        break
    ll1=input().split()
    ll2=input().split()
    for i in ll1:
        l1.append(i)
    for i in ll2:
        l2.append(i)
    lcs=[ [0]*101 for _ in range(101)]
    for i in range(1, N2+1):
        for j in range(1, N1+1):
            if l1[j-1]==l2[i-1]:
                lcs[i][j]=lcs[i-1][j-1]+1
            else:
                lcs[i][j]=max(lcs[i-1][j], lcs[i][j-1])
    print( "Twin Towers #{}".format(num))
    num+=1
    print( "Number of Tiles : {}".format(lcs[N2][N1]) )
    print( '')

File: uva_10066.cpp
# include <iostream>
# include <vector>
using namespace std;
int main(void){
    int N1=0, N2=0;
    int num=0;
    while(cin>>N1>>N2){
        vector<int> L1, L2;
        if (!N1||!N2)
            break;
        L1.resize(N1);
        L2.resize(N2);
        for (int i=0; i<N1; ++i)
            cin >> L1[i];
        for (int i=0; i<N2; ++i)
            cin >> L2[i];
        if (N1>N2){
            swap(L1, L2), swap(N1, N2);
        }
        int LCS[101][101]={0};
        for (int i=1; i<=N2; ++i){
            for (int j=1; j<=N1; ++j)
                if (L1[j-1]==L2[i-1])
                    LCS[i][j]=LCS[i-1][j-1]+1;
                else 
                    LCS[i][j]=max(LCS[i-1][j], LCS[i][j-1]);
        }
        cout << "Twin Towers #" << ++num << endl;
        cout << "Number of Tiles : " << LCS[N2][N1] << endl;
        cout << "\n";
    }
    return 0;
}